tools/add.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+# Shell script equivalent of Metamath ADD tool
+# Mario Carneiro 5-Jan-2022
+
+usage() {
+  cat << HELP
+ADD - Add a specified string to each line in a file
+
+This command adds a character string prefix and/or suffix to each
+line in a file.
+Syntax:  add.sh BEGSTR ENDSTR < INFILE > OUTFILE
+HELP
+}
+
+if [ $# -eq 1 ] && [ "$1" = "-h" ]; then usage; exit; fi
+if [ $# -ne 2 ]; then usage; exit 1; fi
+
+while read line; do echo $1${line}$2; done
```

Comment:
The script won't stop on
./add.sh a b

Reply:
You must provide two parameters, plus you should pipe it the input file if you don't want it to sit there doing nothing. To skip one or the other, you pass an empty string, i.e. `add.sh pfx "" < in.txt` will prepend `pfx` on every line. I believe this matches the metamath behavior, except that it will go through an interactive prompt to get you to finish the invocation if you don't pass enough parameters the first time.

Reply:
As mentioned in the PR comment, these are all "pipe" style programs like `cat`, which is useful for chaining but means that using the same file as input and output does not work. There is an additional utility `inplace.sh` provided for this purpose: `inplace.sh in.txt add.sh pfx ""` will call `add.sh pfx ""` with an in-memory copy of `in.txt` and write the result back to `in.txt`.

Reply:
suggest to remove the or between prefix and suffix and be more explicit about BLANK

Reply:
ack